### PR TITLE
Writing tests for enumerable methods

### PR DESCRIPTION
**Pull request by MarkoNS1990**
In this project, we were making tests for our enumerable methods by using rspec.

### Testing instructions

- `rspec spec/data_type_test.rb` testing what type of class is being returned
- `rspec spec/my_all_test.rb` testing of #my_all
- `rspec spec/my_any_test.rb` testing of #my_any
- `rspec spec/my_each_test.rb` testing of #my_each
- `rspec spec/my_inject_test.rb` testing of #my_inject
- `rspec spec/my_map_test.rb` testing of #my_map
- `rspec spec/my_select_test.rb` testing of #my_select

### CODE REVIEW

**Review by mricanho — CHANGES_REQUESTED** [review]
# 🔴  Changes Required 🔴  

Hello @MarkoNS1990 @marijanbrvar, you did a good job with your rspecs🔥, just a few changes are required:

- [x] **Add a section in your README file with instructions on how to run your rspecs**
- [x] **Right now your tests are not running, check that, please** 😞 
- [x] **All the enumerable tests must be in one file, right now you have one file per each enumerable case**
- [x] **You missed the multiply_els's enumerable, please add it to your tests**
- [x]  **Add more tests that cover [negative ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png)scenarios** 

You can contact me on slack if you have any questions just search for me as @ Miguel Ricaño.

### 💻 👩🏽‍💻  Happy Coding!👨🏽‍💻  🖥️

File: spec/enumerables_test.rb [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  it 'should return true when number is passed as an argument if none of the collection includes given number' do
+    expect(words.my_none?(5) == words.none?(5)).to eql(true)
+  end
+end
+
+#   #my_select tests
+describe '#my_select' do
+  it 'should return true if block returns false or nil' do
+    expect([1, 2, 3, 4, 5].my_select(&:odd?) == [1, 2, 3, 4, 5].select(&:odd?)).to eql(true)
+    expect([1, 2, 3, 4, 5].my_select(&:odd?) == [2, 4, 5, 7]).to eql(false)
```
Comment (Dipeshtwis) [inline spec/enumerables_test.rb]: There is two expected behavior for one `it` block.

File: spec/enumerables_test.rb [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  end
+end
+
+#   multiply_els test
+describe '#multiply_els' do
+  it 'should return calculation based on #my_inject' do
+    expect(multiply_els([2, 4, 5]) == 40).to eql(true)
+    expect(multiply_els([2, 4, 5]) == 50).to eql(false)
```
Comment (Dipeshtwis) [inline spec/enumerables_test.rb:241]: There is two expected behavior for one `it` block.

File: spec/enumerables_test.rb [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+describe '#my_each_with_index' do
+  it 'should return true if it is an Enumerator' do
+    expect([1, 2, 3].my_each_with_index.instance_of?(Enumerator)).to eql(true)
+  end
+end
+
+describe '#my_select' do
+  it 'should return true if it is an Enumerator' do
+    expect([1, 2, 3].my_select.instance_of?(Enumerator)).to eql(true)
+  end
```
Comment (Dipeshtwis) [inline spec/enumerables_test.rb]: There is only one test for `my_select`, `my_each_with_index`, `my_each`. Please test it for more cases.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
# Status: ♻️ Changes Required ♻️
## Hi @marijanbrvar and @MarkoNS1990 , I'm Dipesh your reviewer for this PR.
You have fixed almost every request and all your test is passing at the moment. But in order to approve we have to work on some of the required changes mentioned below.

## For Approval
- [ ] You should have one `it` block for each expected behaviour. I have commented below. Kindly look at that.
- [ ] Some of the method have only one test. Please do it for more cases.

## [Optional]
- [ ] After running the test, I got the warning. 
      -  ./enumerables.rb:72: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead.
    It's better to fix the warning.
---------

Code review by **Dipesh Kumar**

(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
Happy Coding
[Dipesh Kumar](https://deeptwist.tk)
[Twitter](https://twitter.com/97deepeshkumar)

File: spec/enumerables_test.rb [hunk 4]
```diff
@@ -0,0 +1,261 @@
+require_relative '../enumerables'
+
+# Data_type tests
+describe '#my_each' do
+  it 'should return true if it is an Enumerator' do
+    expect([1, 2, 3].my_each.instance_of?(Enumerator)).to eql(true)
```
Comment (Dipeshtwis) [inline spec/enumerables_test.rb]: One describe block for `my_each` here. another is on line 121.

File: spec/enumerables_test.rb [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+  end
+end
+
+describe '#my_each_with_index' do
+  it 'should return true if it is an Enumerator' do
+    expect([1, 2, 3].my_each_with_index.instance_of?(Enumerator)).to eql(true)
```
Comment (Dipeshtwis) [inline spec/enumerables_test.rb]: One describe block for `my_each_with_index` here. another is on line 140.

File: spec/enumerables_test.rb [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+  end
+end
+
+describe '#my_select' do
+  it 'should return true if it is an Enumerator' do
+    expect([1, 2, 3].my_select.instance_of?(Enumerator)).to eql(true)
```
Comment (Dipeshtwis) [inline spec/enumerables_test.rb]: One describe block for `my_select` here. another is on line 244.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
# Status: ♻️ Changes Required ♻️
## Hi @marijanbrvar  and @MarkoNS1990, I'm Dipesh your reviewer for this PR.
You have fixed almost every request and all your test is passing. But in order to approve we have to work on some of the required changes mentioned below.

## For Approval
- [ ] Specs for each method should be wrapped in one describe block. For now, you have two describe block for one method. i.e. `my_each` `my_each_with_index` and `my_select`. I have mentioned below.


Code review by **Dipesh Kumar**

(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
Happy Coding
[Dipesh Kumar](https://deeptwist.tk)
[Twitter](https://twitter.com/97deepeshkumar)

**Review by Zappat0n — APPROVED** [review]
# Status: Approved :tada: :tada:
Great work implementing the changes required by @Dipeshtwis.

You should take this into consideration:

[OPTIONAL] Disabling cops multiple lines in the same file, disabling multiple cops at the top of a document, or disabling easy-to-fix cops such as Layout/LineLength, Style/GuardClause, or Style/GlobalVars should be avoided.

[OPTIONAL] You should use case statements instead of chained if..else statements to decrease the complexity in your logic.

Keep up the good work!.

You can contact me through Slack @Zappat0n for any further clarification.
